src/main.rs

```diff
@@ -61,7 +62,7 @@ where
 
 #[inline(always)]
 #[get("/version")]
-async fn version(state: Data<ServerActor<repository::Warehouse>>) -> Result<impl Responder> {
+async fn version(state: Data<ServerActor<WarehouseMongo>>) -> Result<impl Responder> {
```

Comment:
Suggestion: Using tool's name in structure name creates tight-coupling that will affect the rest of the app. I'd change it to just "Warehouse" as it was before, and inject Mongo into warehouse.

It gives us possibility to change it when we learn that there is a better solution.

Reply:
Ok, thanks. I think it makes sense.